### DISCUSSION TITLE
Exception when building MsSqlTestContainer

### DISCUSSION THREAD

**Discussion by siguro58** [testcontainers/testcontainers-dotnet#543] [Q&A] [answered]
When instantiating TestcontainersBuilder like this:
`var builder = new TestcontainersBuilder<MsSqlTestcontainer>()`

this exception is thrown:

> System.TypeInitializationException: 'The type initializer for 'DotNet.Testcontainers.Configurations.TestcontainersSettings' threw an exception.'
> Inner exception
> InvalidOperationException: Sequence contains no elements

Below is an excerpt of the stack trace:

> at DotNet.Testcontainers.Configurations.TestcontainersSettings.get_OS()
>    at DotNet.Testcontainers.Builders.TestcontainersBuilder`1..ctor()

The code is built with .net core 6
Any idea what is causing the exception?

**Comment by siguro58** [comment, marked as answer]
Found the answer, setting the DOCKER_HOST environment variable worked.

**Comment by siguro58** [comment]
Except now I get an error when executing `builder.Build()`: Image can not be null. (Parameter 'Image')
Tried setting the image with `builder.WithImage("mcr.microsoft.com/mssql/server:2017-CU28-ubuntu-16.04")`
but get the same error

**Reply by HofmeisterAn** [reply]
Which Docker engine do you use? Can you add the `docker info` output?
